src/app/jscaip/state/HexagonalGameState.ts

```diff
@@ -49,20 +49,14 @@ export abstract class HexagonalGameState<P extends NonNullable<Comparable>> exte
                 return false;
             }
         }
-        for (const coord of this.allCoords()) {
-            if (equal(this.getPieceAt(coord), other.getPieceAt(coord)) === false) {
-                return false;
+        let isEqual: boolean = true;
+        this.forEachCoord((coord: Coord, content: P) => {
+            if (equal(content, other.getPieceAt(coord)) === false) {
+                isEqual = false;
+                return;
```

Comment:
tu viens de rajouter des tours de boucles inutiles ! (et le return ici est inutile également)
L'ancienne formulation s'arrêtait dès qu'on trouvait un faux, ici on fait TOUS les coords no matter what.
Peut-être il nous faut une variante de forEachCoord qui est genre findCoordSuchThat(... === false)

Reply:
and thus findMatchingCoord is born